services/src/util/apidoc.rs

```diff
@@ -148,19 +161,31 @@ impl Modify for TransformSchemasWithTag {
                 let Some(variant_tag) = Self::get_variant_tag(item, discriminator) else {
                     continue 'outer;
                 };
-                let variant_schema_name = match Self::get_base_type_name(item) {
-                    Some(base_type) => format!(
-                        "{}With{}",
-                        base_type,
-                        Self::uppercase_first_letter(discriminator),
-                    ),
-                    None => format!(
-                        "{}{}",
-                        Self::uppercase_first_letter(variant_tag),
-                        schema_name
-                    ),
+
+                let variant_schema_name = match Self::get_title(item) {
+                    // Always prefer custom title as variant name.
```

Comment:
is this a proper use of the `title` field or is it a "hack" to make names configurable without adding new functionality to utoipa?

Reply:
It is a bit of a hack, but others use it too. I don't have the source anymore, but I read on websites that some code generators base the generated class names on the title attribute.

Reply:
ok, can you then add a comment about this in the code about: what the original purpose of this field is and how we (ab)use it?

Reply:
Using `#[schema(rename = "custom_value")]` is effectively the same as renaming the variant in the Rust code. This leads to no additional information for the enum transformation code.

When that attribute is employed, the default naming strategy is executed:
1. Has single tuple payload? Use payload name as schema name.
2. Otherwise: `{Enum-Name}{Variant-Name}

This means the following:
- in case 1: The renaming-attribute does nothing
- in case 2: rename-attribute value not used as full name

For example,

```rust
enum DataId {
    #[schema(rename = "InternalDataId")]
    Internal {
        dataset_id: DatasetId,
    }
}
```

leads to a type named `DataIdInternalDataId`, because it falls under case 2.

Because of that `#[schema(rename = "custom_value")]` can not be used for the desired purpose.

Reply:
I added a comment here: [15da92f](https://github.com/geo-engine/geoengine/pull/953/commits/15da92fae8d7436bebe933d9fe11082e6b5e82f2)